tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -2638,6 +2638,8 @@ LogicalResult ConvertTFLSliceOp::matchAndRewrite(
     Operation* op, PatternRewriter& rewriter) const {
   auto tfl_slice_op = cast<TFL::SliceOp>(op);
 
+  auto input = tfl_slice_op.getInput();
+  ShapedType input_type = input.getType().dyn_cast<ShapedType>();
```

Comment:
Switch to cast (and new preferred form is cast<ShapedType>(input.getType())) as value is used without checking in 2657 (and also is verified to be shaped type by verifier).

Reply:
RankedTensorType differs a bit from ShapedType as in mlir/lite/ir/tfl_ops.td you'll see the operands marked as being shaped always, but ranked only sometimes. I'm fine with consistent but expectation is this runs on a verified form.

Reply:
> RankedTensorType differs a bit from ShapedType as in mlir/lite/ir/tfl_ops.td you'll see the operands marked as being shaped always, but ranked only sometimes. I'm fine with consistent but expectation is this runs on a verified form.

Thanks for pointing me to tfl_ops.td. The SliceOp seems to expect a ranked input operand [1] so I've simplified the code and cast directly to RankedTensorType.

[1] https://github.com/tensorflow/tensorflow/blob/c7a8807051e717be9b182f94fd40f0088a7fabc3/tensorflow/compiler/mlir/lite/ir/tfl_ops.td#L2345